bundles/action/org.openhab.action.twitter/src/main/java/org/openhab/action/twitter/internal/Twitter.java

```diff
@@ -57,21 +64,104 @@ public static boolean sendTweet(@ParamDoc(name = "tweetTxt", text = "the Tweet t
             logger.debug("Twitter client is disabled > execution aborted!");
             return false;
         }
-
+        
+        return true;
+    }
+    
+    /**
+     * Internal method about sending a tweet, with or without image
+     * 
+     * @param tweetTxt the Tweet to send
+     * @param fileToAttach the file to attach. May be null if no attached file.
+     * 
+     * @return <code>true</code>, if sending the tweet has been successful and
+     *         <code>false</code> in all other cases.
+     */
+    private static boolean sendTweet(final String tweetTxt, final File fileToAttach) {
+        // abbreviate the Tweet to meet the 140 character limit ...
+        final String abbreviatedTweetTxt = StringUtils.abbreviate(tweetTxt, CHARACTER_LIMIT);
         try {
-            // abbreviate the Tweet to meet the 140 character limit ...
-            tweetTxt = StringUtils.abbreviate(tweetTxt, CHARACTER_LIMIT);
             // send the Tweet
-            Status status = client.updateStatus(tweetTxt);
-            logger.debug("Successfully sent Tweet '{}'", status.getText());
+            final StatusUpdate status = new StatusUpdate(abbreviatedTweetTxt);
+            if (fileToAttach != null && fileToAttach.isFile()) {
+            	status.setMedia(fileToAttach);
+            }
+            final Status updatedStatus = client.updateStatus(status);
+            logger.debug("Successfully sent Tweet '{}'", updatedStatus.getText());
             return true;
         } catch (TwitterException e) {
-            logger.error("Failed to send Tweet '" + tweetTxt + "' because of: " + e.getLocalizedMessage());
+            logger.error("Failed to send Tweet '" + abbreviatedTweetTxt + "' because of: " + e.getLocalizedMessage());
```

Comment:
This should be a WARN instead of an ERROR.
Also, need to use parameterized logging instead of string concatenation.